### PR TITLE
48532_trainee_view_subject_progress

### PR DESCRIPTION
**Pull request by yasin-buet**
screenshot from 2015-12-14 16 23 20

### CODE REVIEW

**Comment by asifbuetcse** [comment]
ok

File: app/Http/Controllers/Trainee/SubjectsController.php [hunk 1]
```diff
@@ -18,7 +18,20 @@ public function index()
...
+                                ->where('task_user.status', 1)
+                                ->where('task_user.course_id', $course->id)
+                                ->where('task_user.subject_id', $subject->id)
+                                ->count();
+            $tasksNotFinished = \Auth::user()->tasks()
+                                    ->where('task_user.status', 0)
+                                    ->where('task_user.course_id', $course->id)
+                                    ->where('task_user.subject_id', $subject->id)
+                                    ->count();
+            $subjectProgress[$index] = ($tasksFinished / ($tasksFinished + $tasksNotFinished)) * 100;
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Trainee/SubjectsController.php]: Calculate progress inside model is better approach.
http://laravel.com/docs/5.1/eloquent-mutators
Read this documentation and define function `getProgressAttribute`.
After that, you can call `$subject->progress` easily.

File: app/Subject.php [hunk 2]
```diff
@@ -11,4 +11,19 @@ public function tasks()
     {
         return $this->hasMany('App\Task');
     }
+    public function getProgressAttribute($value) {
+        $course = \Auth::user()->courses()->where('course_user.is_currently_enrolled', 1)->first();
```
Comment (dinhhoanglong91) [inline app/Subject.php]: Sorry I found out a problem.
Every time you call `$subject->progress`, this code is run. It select current course from DB.
This process is already done when you find `$course` in controller.
So, in order to remove this repeated process, you should pass `$courseId` to this function.
But, in this case, we can't use mutator.
Therefore, let's try simple function
```php
// Model
public function progress($courseId)
{
    // Calculate $progress
}
/// View
$subject->progress($courseId);
```

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**